maven.nix

```diff
@@ -0,0 +1,27 @@
+{ lib, fetchurl, linkFarm}:
+with lib;
+# Create a maven environment from the output of the mvn2nix command
+# the resulting store path can be used as the a .m2 repository for subsequent
+# maven invocations.
+# ex.
+# 	mvn package --offline -Dmaven.repo.local=${dependencies}
+{
+	name ? "buildMavenRepsitory",
+	dependencies ? import ./dependencies.nix,
```

Comment:
Note that `./dependencies.nix` here will always refer to the one inside this repository, not one that is reachable relative to from where the function is called. I think the latter would be a more useful default, just not sure how to (idiomatically) achieve that. @roberth do you know?

Reply:
I was trying to copy **bundix** and it's use of finding the **gemset.nix**

Looks like you either have to provide the dependency file (i.e. _dependency.nix_) or the directory which it then uses to resolve the default name.

I'll just make it non-default for now.

Reply:
Just checked the implementation of `bundleEnv`. Looks like it also requires you to pass the file explicitly via the `gemset` argument or alternatively, pass a directory explicitly via `gemdir` under which `gemset.nix` will then be looked up. See https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/ruby-modules/bundled-common/functions.nix#L25-L27 for the relevant code.

Reply:
Yes; I saw that too. For now I made it explicit.

I did notice a few more things I need to augment:

1. I need to collect the plugins used in the POM file
2. I need to include the pom for each JAR (or other packaging type)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The current third party library I'm using doesn't have this capability so i'm exploring other options such as using _aether_ (Maven's underlying library); although there's not much documentation or examples.